pkg/webhook/validatingwebhook.go

```diff
@@ -0,0 +1,136 @@
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ValidatingWebhookConfigurationBuilder provides struct for ValidatingWebhookConfiguration object
+// which contains connection to cluster and ValidatingWebhookConfiguration definition.
+type ValidatingWebhookConfigurationBuilder struct {
+	// ValidatingWebhookConfiguration definition. Used to create ValidatingWebhookConfiguration object.
+	Definition *admregv1.ValidatingWebhookConfiguration
+	// Created ValidatingWebhookConfiguration object.
+	Object *admregv1.ValidatingWebhookConfiguration
+	// Used in functions that define or mutate ValidatingWebhookConfiguration definitions.
+	// errorMsg is processed before ValidatingWebhookConfiguration object is created.
+	errorMsg string
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+}
+
+// PullValidatingWebhookConfiguration pulls existing ValidatingWebhookConfiguration from cluster.
+func PullValidatingWebhookConfiguration(apiClient *clients.Settings, name string) (
+	*ValidatingWebhookConfigurationBuilder, error) {
+	glog.V(100).Infof("Pulling existing ValidatingWebhookConfiguration name %s from cluster", name)
+
+	builder := ValidatingWebhookConfigurationBuilder{
+		apiClient: apiClient,
+		Definition: &admregv1.ValidatingWebhookConfiguration{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the ValidatingWebhookConfiguration is empty")
+
+		builder.errorMsg = "ValidatingWebhookConfiguration 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("ValidatingWebhookConfiguration object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Exists checks whether the given ValidatingWebhookConfiguration object exists in the cluster.
+func (builder *ValidatingWebhookConfigurationBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get returns ValidatingWebhookConfiguration object.
+func (builder *ValidatingWebhookConfigurationBuilder) Get() (*admregv1.ValidatingWebhookConfiguration, error) {
+	if valid, err := builder.validate(); !valid {
+		return &admregv1.ValidatingWebhookConfiguration{}, err
+	}
+
+	validatingWebhookConfiguration := &admregv1.ValidatingWebhookConfiguration{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, validatingWebhookConfiguration)
+
+	if err != nil {
+		return &admregv1.ValidatingWebhookConfiguration{}, err
+	}
+
+	return validatingWebhookConfiguration, err
+}
+
+// Delete removes a ValidatingWebhookConfiguration from a cluster.
+func (builder *ValidatingWebhookConfigurationBuilder) Delete() (*ValidatingWebhookConfigurationBuilder, error) {
```

Comment:
ditto, what about `DeleteAndWait`

Reply:
As Nikita menshioned no need for this resource